lib/galaxy/webapps/galaxy/controllers/admin.py

```diff
@@ -549,246 +540,203 @@ def users_list( self, trans, **kwd ):
+                elif operation == 'undelete':
+                    message = self._undelete_quota( quotas )
+                elif operation == 'purge':
+                    message = self._purge_quota( quotas )
+                elif operation == 'unset as default':
+                    message = self._unset_quota_default( quotas[ 0 ] )
+            except ActionInputError as e:
+                message, status = ( e.err_msg, 'error' )
+        if message:
+            kwargs[ 'message' ] = util.sanitize_text( message )
+            kwargs[ 'status' ] = status or 'done'
+        kwargs[ 'dict_format' ] = True
         return self.quota_list_grid( trans, **kwargs )
 
-    @web.expose
+    @web.expose_api
     @web.require_admin
-    def create_quota( self, trans, **kwd ):
-        params = self.get_quota_params( kwd )
-        if params.get( 'create_quota_button', False ):
+    def create_quota( self, trans, payload=None, **kwd ):
+        if trans.request.method == 'GET':
+            all_users = []
+            all_groups = []
+            for user in trans.sa_session.query( trans.app.model.User ) \
+                                        .filter( trans.app.model.User.table.c.deleted == false() ) \
+                                        .order_by( trans.app.model.User.table.c.email ):
+                all_users.append( ( user.email, trans.security.encode_id( user.id ) ) )
+            for group in trans.sa_session.query( trans.app.model.Group ) \
+                                         .filter( trans.app.model.Group.table.c.deleted == false() ) \
+                                         .order_by( trans.app.model.Group.table.c.name ):
+                all_groups.append( ( group.name, trans.security.encode_id( group.id ) ) )
+            default_options = [ ( 'No', 'no' ) ]
+            for typ in trans.app.model.DefaultQuotaAssociation.types.__dict__.values():
+                default_options.append( ( 'Yes, ' + typ, typ ) )
+            return {    'title'  : 'Create Quota',
+                        'inputs' : [{
+                            'name'    : 'name',
+                            'label'   : 'Name'
+                        }, {
+                            'name'    : 'description',
+                            'label'   : 'Description'
+                        }, {
+                            'name'    : 'amount',
+                            'label'   : 'Amount',
+                            'help'    : 'Examples: "10000MB", "99 gb", "0.2T", "unlimited"'
+                        }, {
+                            'name'    : 'operation',
+                            'label'   : 'Assign, increase by amount, or decrease by amount?',
+                            'options' : [ ('=', '=' ), ( '+', '+' ), ( '-', '-' ) ]
+                        }, {
+                            'name'    : 'default',
+                            'label'   : 'Assign, increase by amount, or decrease by amount?',
```

Comment:
this should probably read `Is this quota a default for a class of users (if yes, what type)?`